src/pyfmi/fmi_util.pxd

```diff
@@ -49,6 +50,16 @@ ELSE:
     cdef inline long long os_specific_ftell(FILE *stream):
         return ftello(stream)
 
+cdef class DumpDataFMI3:
+    cdef np.ndarray time_tmp
+    # TODO: Would make more sense if 'model' is declared as FMI3.FMUModelME3
```

Comment:
Why not; this seems to work for FMI2?

Reply:
From a brief look on FMI2; `model` is a also a generic `public object` there, but there is the additional explicit `model_me2` there, with assignment behind an explicit type check. 

We may want to use a similar format here; I don't fully understand the practical implications, but it was motivated by performance if I recall correctly. Could also postpone this until later when we evaluate performance for FMI3.
Either way, maybe could add a comment as reminder?